x-pack/plugins/ml/server/routes/system.ts

```diff
@@ -215,29 +215,34 @@ export function systemRoutes(
     {
       path: '/api/ml/index_exists',
       validate: {
-        body: schema.object({ index: schema.string() }),
+        body: schema.object({ indices: schema.arrayOf(schema.string()) }),
       },
       options: {
         tags: ['access:ml:canAccessML'],
       },
     },
     routeGuard.basicLicenseAPIGuard(async ({ client, request, response }) => {
       try {
-        const { index } = request.body;
+        const { indices } = request.body;
 
         const options = {
-          index: [index],
+          index: indices,
           fields: ['*'],
           ignore_unavailable: true,
           allow_no_indices: true,
         };
 
         const { body } = await client.asCurrentUser.fieldCaps(options);
-        const result = { exists: false };
 
-        if (Array.isArray(body.indices) && body.indices.length !== 0) {
-          result.exists = true;
-        }
+        const result = indices.reduce((acc, cur) => {
+          acc[cur] = { exists: body.indices.includes(cur) };
+          return acc;
+        }, {} as Record<string, { exists: boolean }>);
+        // // const result = { exists: false };
+
+        // if (Array.isArray(body.indices) && body.indices.length !== 0) {
+        //   result.exists = true;
+        // }
```

Comment:
commented code

Reply:
Fixed in https://github.com/elastic/kibana/pull/101037/commits/cfad0ead1352a0f6ac4c70f12efd337ee0eaa1f1